src/martialarts.cpp

```diff
@@ -863,6 +877,24 @@ bool player::can_miss_recovery( const item &weap ) const
     return true;
 }
 
+bool player::is_wall_adjacent() const
```

Comment:
And this function is only used here, so why does it need to be part of the (already very large) `player` class? It would work just fine as a free function taking only the center position as parameter (doesn't even need the other properties of the `player` class).

Reply:
It's also called in melee.cpp. It's part of the `player` class because all martial arts functions need to be on the `player` for the sake accessibility. If I turned it into a free function, I would have to create a wrapper class on the `player` class anyway just to call it outside of martialarts.cpp just like `search_ma_buff_effect` needs wrapper classes to be used.

I could move it to player.cpp for more general use but it still needs to be part of the `player` class for accessibility.

Reply:
> it still needs to be part of the player class for accessibility.

Why? The only thing it needs from `player` is its position, which is public available via `pos()`. No other property is needed.

A function like this would work just as well (and be much more generic as it can be applied to any point on the map):
```C++
bool is_wall_adjacent( const tripoint &center ) {
    for( const tripoint &p : g->m.points_in_radius( center, 1 ) ) {
        if( p != center && g->m.impassable( p ) ) {
            return true;
        }
    }
    return false;
}
```

Reply:
Ah, that function works better than mine! I'll move it to map.cpp since it doesn't need to be on the player anymore.